### DISCUSSION TITLE
Windows setup error "Please put your prompt in the file 'prompt' in the project directory

### DISCUSSION THREAD

**Discussion by JacobRuddiman** [AntonOsika/gpt-engineer#421] [Q&A]
Hi, I'm trying to set up GPT engineer on Windows 11, but keeping getting this issue and I can't figure it out. I have installed all the requirements, and when using python -m gpt_engineer.main prompt it says that I don't have a prompt file in the project directory which I do.
Any ideas would be great.

**Comment by abhising007** [comment]
I had the same issue. The problem is path misalignment. You need to execute the command from project root folder for it to work.
e.g. if you cloned the repo in d:\genAi\gpt-engineer , you need to execute following command from d:\genAi\gpt-engineer director:
PS d:\genAi\gpt-engineer > **python gpt_engineer\main.py projects\<project name>**

The error was resolved. I am yet to use this project successfully though.

**Reply by efkopru** [reply]
worked for me , thanks!

**Comment by geewhizbang** [comment]
This answer makes no sense whatsoever. I have installed gpt-engineer in c:\source\gpt-engineer

I have a blank folder c:\source\myGPT

So this is supposed to work: 

     gpt-engineer source\myGPT

whether I'm in either folder c:\source\myGPT or c:\source\gpt-engineer it gives the assertion error that makes no sense at all:

      AssertionError: Please put your prompt in the file `prompt` in the project directory

**Reply by vVv-Keys** [reply]
To resolve this issue, follow these steps:

Navigate to either the c:\source\myGPT or c:\source\gpt-engineer directory, depending on your preference.

Create a new file called prompt within the chosen directory.

Open the prompt file and enter your desired prompt text. This prompt will be used by the code you're running.

Save the prompt file.

After completing these steps, try running the gpt-engineer command again. It should now read the prompt from the prompt file without triggering the assertion error.

**Reply by geewhizbang** [reply]
Yes, I eventually figured that out. Thanks

**Comment by jenspacito** [comment]
Yesterday again and again was the energy out, maybe must look. No Problem. But thank you so much. Its a really good family and community here. I like.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by vVv-Keys** [comment]
> Hi, I'm trying to set up GPT engineer on Windows 11, but keeping getting this issue and I can't figure it out. I have installed all the requirements, and when using python -m gpt_engineer.main prompt it says that I don't have a prompt file in the project directory which I do. Any ideas would be great.

To resolve this issue, follow these steps:

Navigate to either the c:\source\myGPT or c:\source\gpt-engineer directory, depending on your preference.

Create a new file called prompt within the chosen directory.

Open the prompt file and enter your desired prompt text. This prompt will be used by the code you're running.

Save the prompt file.

After completing these steps, try running the gpt-engineer command again. It should now read the prompt from the prompt file without triggering the assertion error.

**Reply by jgovednik** [reply]
Is there a functional reason the creation of a prompt file isnt the first step when one is not detected? 
Seems like a massive misstep for the UsereXperience to throw an error during a logical starting point (running against an empty folder is use case 0). Especially when the error message does not sufficiently describe why its being thrown and/or how to resolve the error, which help command to run for more info, or a website with an error ID to enable the user to lookup, etc. Expecting a "prompt" file to exist may be YOUR expectation, but its not actually listed as a requirement in the --help (nor, fyi, is the github link, which is not required to install via pip, and should probably be linkbacked within the actual module)

**Comment by vVv-Keys** [comment]
if you have any more questions feel free to let me know or if it's still not working lmk the error and I'll try to figure it out 🗡️

**Comment by skamenan7** [comment]
I got it working like below

c:\git\gpt-engineer-stuff\gpt-engineer(main -> origin)
(GPT_Engineer) λ python gpt_engineer\main.py projects\neural-network
